content/en/blog/_posts/2023-03-08-memory-qos-cgroups-v2/index.md

```diff
@@ -0,0 +1,260 @@
+---
+layout: blog
+title: 'Alpha Support for Quality-of-Service for Memory Resources'
+date: 2023-03-08
+slug: qos-memory-resources
+---
+
+**Authors:** Dixita Narang (Google)
+
+Kubernetes v1.27, released in April 2023, introduced changes to an alpha feature,
+Memory QoS, that improves memory management capabilites in Linux nodes. 
+
+Support for Memory QoS feature was initially added in Kubernetes v1.22, and later some 
+[limitations](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2570-memory-qos#reasons-for-changing-the-formula-of-memoryhigh-calculation-in-alpha-v127) 
+around the formula for calculating memory.high were identified. These limitations are 
+addressed in Kubernetes v1.27.
+
+## Background
+
+Kubernetes allows you to optionally specify how much of each resources a container needs
+in the Pod specification. The most common resources to specify are CPU and Memory. 
+
+For example, a Pod spec with container resources could look like as follows:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example
+spec:
+  containers:
+  - name: nginx
+    resources:
+      requests:
+        memory: "64Mi"
+        cpu: "250m"
+      limits:
+        memory: "64Mi"
+        cpu: "500m"
+```
+
+* `spec.containers[].resources.requests`
+
+  When you specify the resource request for containers in a Pod, the [kube-scheduler](/docs/concepts/scheduling-eviction/kube-scheduler/#kube-scheduler)
+  uses this information to decide which node to place the Pod on. Each node has a 
+  maximum capacity for each of the resource types: the amount of CPU and memory it 
+  can provide for Pods. The scheduler ensures that, for each resource type, the sum
+  of the resource requests of the scheduled containers is less than the capacity of
+  the node.
```

Comment:
This is true, but it might be worth to clarify: the node's advertise capacity and allocatable where allocatable = capacity - reserved. The scheduler will only consider allocatable when making scheduling decisions. I don't think it's worth going too much about this in the blog since it's not relevant, but maybe worth to replace "capacity" with total allocatable here.